qp_klp/klp_util.py

```diff
@@ -1,5 +1,35 @@
 from os.path import join
 import pandas as pd
+from json import dumps
+
+
+def update_blanks_in_qiita(sifs, qclient):
+    for sif_path in sifs:
+        # get study_id from sif_file_name ...something_14385_blanks.tsv
+        study_id = sif_path.split('_')[-2]
+
+        # SIFs only contain BLANKs. Get the list of potentially new BLANKs.
+        blanks = pd.read_csv(sif_path, delimiter='\t')['sample_name']
+
+        # Prepend study_id to make them compatible w/list from Qiita.
+        blanks = [f'{study_id}.{x}' for x in blanks]
+
+        # Get list of BLANKs already registered in Qiita.
+        from_qiita = qclient.get(f'/api/v1/study/{study_id}/samples')
+        from_qiita = [x for x in from_qiita if
+                      x.startswith(f'{study_id}.BLANK')]
+
+        # Generate list of BLANKs that need to be ADDED to Qiita.
+        new_blanks = (set(blanks) | set(from_qiita)) - set(from_qiita)
+
+        if len(new_blanks):
```

Comment:
Maybe I'm wrong but I think this will add dummy data to all columns of the blanks vs. _only_ dummy data for those columns that do _not_ exist in the local blanks; right?

Reply:
If I understand your question correctly, no I don't think it will add dummy data over and above what is already there. I've confirmed new_blanks will just be the subset of BLANKs found in the SIFs but not found in Qiita. Those BLANKs will be inserted w/dummy data, but the categories are all pre-existing so no new columns are being added to the study. The existing BLANKs in Qiita should already have (possibly dummy) data associated with them and they should not be overwritten by this PATCH, since it's not a POST.